### PR TITLE
[CONFIGURE] Add VS 16.0 support, simplify generator selection

### PR DESCRIPTION
**Pull request by tambry**
## Purpose

This simplifies the Visual Studio generator selection logic and adds support for Visual Studio 16.0 (aka 2019).

### CODE REVIEW

File: configure.cmd [hunk 1]
```diff
@@ -140,48 +140,13 @@ REM Parse command line parameters
                 set VS_VERSION=%3
                 echo Visual Studio Environment set to !BUILD_ENVIRONMENT!!VS_VERSION!-!ARCH!
             )
-            if "!VS_VERSION!" == "10" (
```
Comment (HBelusca) [inline configure.cmd]: Does this also work for VS 2010?

Comment (tambry) [reply configure.cmd]: I see no reason it shouldn't. CMake supports the -A option for `VS_VERSION>=8`.  
That said, I don't have Visual Studio 2010 on hand to test.

**Comment by tkreuzer** [comment]
Did you test x64 build?

**Comment by tambry** [comment]
@tkreuzer I did. It works.

**Comment by binarymaster** [comment]
Huh? What happened? Why closed? 😳

**Comment by tambry** [comment]
No one's bothered to review/merge this over a month later, so it's unlikely it'll ever be (same for my 2 other PRs).  
Might as well keep the PR list cleaner, although it's already littered with tons of PRs in a similar state.

**Comment by Extravert-ir** [comment]
Hey, as you may have noticed, PRs are sometimes being reviewed for quite some time in this project.
So "unlikely it'll ever be" part is not true. But for closed PR it's true. Keeping PR list cleaner is our responsibility, not yours

**Comment by tambry** [comment]
That's true. However, I'm no longer interested in contributing partly as a result and I'd rather not have effort wasted on reviewing this if I'm unlikely to address any comments received.

**Comment by blackcrack** [comment]
this is what happens when people are disappointed
 ..

**Comment by HBelusca** [comment]
Re-opening as I'm going to deal with it.

**Comment by HBelusca** [comment]
Committed (with fix for VSSolution support), thanks! And sorry for the delay.

**Pull request merged**